layer2/lib/chusai_common/error.ml

```diff
@@ -0,0 +1,162 @@
+(* MIT License
+
+   Copyright (c) 2022 Marigold <contact@marigold.dev>
+
+   Permission is hereby granted, free of charge, to any person obtaining a copy
+   of this software and associated documentation files (the "Software"), to deal in
+   the Software without restriction, including without limitation the rights to
+   use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+   the Software, and to permit persons to whom the Software is furnished to do so,
+   subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in all
+   copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+   SOFTWARE. *)
+
+open Tezos_base.TzPervasives
+open Chusai_tezos
+
+(* Rexport the type error as an extensible variant. *)
+type t = Tezos_base.TzPervasives.error = ..
+
+(* A dummy error for uncatched error. *)
+type t += Chusai_unknown_error of string
+
+let () =
+  register_error_kind
+    ~id:"chusai.unknown_error"
+    ~title:"unexpected/unknown error"
+    ~description:"An unknown error occurred"
+    ~pp:(fun ppf -> Format.fprintf ppf {|Unknown error with message "%s"|})
+    `Permanent
+    Data_encoding.(obj1 (req "message" string))
+    (function
+      | Chusai_unknown_error message -> Some message
+      | _ -> None)
+    (fun message -> Chusai_unknown_error message)
+;;
```

Comment:
What would you see as additional documentation? It seems to me that it is quite conventional? You need to add an extension to the `Error` variant, then provide its registration.

Reply:
Naming convention, maybe a simple explanation for last 3 args, although I admit I only skimmed the error monad doc, and didn't take the time to fully understand it. 
https://ocaml.org/p/tezos-error-monad/13.0/doc/Tezos_error_monad/TzCore/index.html

It would probably be better to try and add errors when relevant, to avoid reusing existing ones and diluting their meaning. This is the only reason I mentioned it.

Reply:
I had a small explanation in the Error module.